packages/hash/shared/src/save.ts

```diff
@@ -269,20 +268,19 @@ const calculateSaveActions = async (
       const oldChildEntityForBlock = savedEntity.blockChildEntity;
 
       if (
-        oldChildEntityForBlock?.entityId !== newChildEntityForBlock?.entityId
+        oldChildEntityForBlock?.metadata.editionId.baseId !==
+        newChildEntityForBlock?.metadata.editionId.baseId
       ) {
-        if (!newChildEntityForBlock?.entityId) {
+        if (!newChildEntityForBlock?.metadata.editionId.baseId) {
           // this should never happen because users select new child entities from API-provided entities.
           // if this errors in future, it's because users are choosing locally-created but not yet db-persisted entities
```

Comment:
These sorts of things make me wonder if we should be comparing editionId instead of baseId? Because that contains the version

Reply:
I think it may make sense for this case to use the editionId (which would mean checking version because we'd need deep equality otherwise) as this deals with swapping out blocks. Not sure if it changes anything as we would just be swapping out old versions (which would be done anyway when refetching the page)